osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -237,18 +259,37 @@ public void SetHyperDashState(double modifier = 1, float targetPosition = -1)
 
                 if (!wasHyperDashing)
                 {
-                    this.FadeColour(Color4.OrangeRed, hyper_dash_transition_length, Easing.OutQuint);
-                    this.FadeTo(0.2f, hyper_dash_transition_length, Easing.OutQuint);
                     Trail = true;
 
-                    var hyperDashEndGlow = createAdditiveSprite();
-
+                    var hyperDashEndGlow = createAdditiveSprite(endGlowSprites);
                     hyperDashEndGlow.MoveToOffset(new Vector2(0, -10), 1200, Easing.In);
                     hyperDashEndGlow.ScaleTo(hyperDashEndGlow.Scale * 0.95f).ScaleTo(hyperDashEndGlow.Scale * 1.2f, 1200, Easing.In);
                     hyperDashEndGlow.FadeOut(1200);
                     hyperDashEndGlow.Expire(true);
                 }
             }
+
+            updateCatcherColour();
+        }
+
+        private void updateCatcherColour()
+        {
+            const float hyper_dash_transition_length = 180;
+
+            if (HyperDashing)
+            {
+                this.FadeColour(hyperDashColour == DefaultHyperDashColour ? Color4.OrangeRed : hyperDashColour, hyper_dash_transition_length, Easing.OutQuint);
+                this.FadeTo(0.2f, hyper_dash_transition_length, Easing.OutQuint);
+            }
+            else
+            {
+                this.FadeColour(Color4.White, hyper_dash_transition_length, Easing.OutQuint);
+                this.FadeTo(1f, hyper_dash_transition_length, Easing.OutQuint);
+            }
```

Comment:
Did you check what are the implications of no longer checking hyper-dash state on the previous frame and skipping fades if there has been no change, as was done previously? The previous way would avoid redundant colour fades/invalidations if consecutive `updateCatcherColour()` calls were made that would not change value of `HyperDashing`.